src/main/java/org/codehaus/plexus/archiver/AbstractArchiver.java

```diff
@@ -1257,4 +1257,37 @@ public String getOverrideGroupName()
     {
         return overrideGroupName;
     }
+
+    @Override
+    public void configureReproducible( Date outputTimestamp )
+    {
+        // 1. force last modified date
+        setLastModifiedDate( convertOutputTimestamp( outputTimestamp ) );
+
+        // 2. sort filenames in each directory when scanning filesystem
+        setFilenameComparator( new Comparator<String>()
+        {
+            @Override
+            public int compare( String s1, String s2 )
+            {
+                return s1.compareTo( s2 );
+            }
+        } );
+
+        // 3. ignore file/directory mode from filesystem, since they may vary based on local user umask
```

Comment:
This can be quite inconvenient if the archive contains executable scripts (not uncommon for Java applications). I think that at least it should be explicitly stated in the configureReproducible java docs - "reproducible entries Unix mode" is not specific enough. Also I was thinking if it would make sense to keep the executable flag. Having a file mask with the executable flag set to true on Unix system is quite uncommon.

Reply:
yes, executable bit is the most wanted one.
Not sure this bit is really managed when running on Windows.
I don't know what precisely to do
I'll merge the current state as it is a good initial step: please help by providing a PR to improve